### PR TITLE
Some work on all Spotify releated issues

### PR DESCRIPTION
**Pull request by svrooij**

### CODE REVIEW

**Comment by svrooij** [comment]
@bencevans do you want me to wait for at least one review, or do you think I can merge these requests myself?

File: lib/sonos.js [hunk 1]
```diff
@@ -704,18 +719,23 @@ Sonos.prototype.playTuneinRadio = function (stationId, stationTitle, callback) {
...
 /**
- * Add a song from spotify to the queue
+ * Add a song from spotify to the queue, just a shortcut to sonos.queue('spotify:track:....')
  * @param  {String}   trackId      The spotify track ID
  * @param  {Function} callback (err, success)
  */
-Sonos.prototype.addSpotify = function (trackId, callback) {
-  var uri = 'x-sonos-spotify:spotify%3atrack%3a' + trackId
-  var meta = '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="00032020spotify%3atrack%3a' + trackId + '" restricted="true"><dc:title></dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON' + this.options.spotify.region + '_X_#Svc' + this.options.spotify.region + '-0-Token</desc></item></DIDL-Lite>'
+Sonos.prototype.addSpotifyTrack = function (trackId, callback) {
```
Comment (bencevans) [inline lib/sonos.js]: Worth taking this op to rename to `queueSpotifyTrack`? Think it's more descriptive or remove altogether and just support spotify tracks URIs via the `queue()`/`play()` functions.

File: lib/sonos.js [hunk 2]
```diff
@@ -704,18 +719,23 @@ Sonos.prototype.playTuneinRadio = function (stationId, stationTitle, callback) {
...
-  this.queue({
-    uri: uri,
-    metadata: meta
-  }, callback)
+/**
+ * Add a album from spotify to the queue, just a shortcut to sonos.queue('spotify:album:....')
+ * @param  {String}   albumId      The spotify album ID
+ * @param  {Function} callback (err, success)
+ */
+Sonos.prototype.addSpotifyAlbum = function (albumId, callback) {
```
Comment (bencevans) [inline lib/sonos.js]: `queueSpotifyAlbum`?

File: lib/sonos.js (hunk 1 above)
Comment (svrooij) [reply lib/sonos.js]: I think removing them would be great. since they are only shortcuts at the moment.....

**Comment by svrooij** [comment]
I will create a new commit removing the spotify specific functions. But what about the radio function?

My idea is to create a `playWithoutQueue` function that could be used for starting the radio stations and that also could be used to play notifications (that would make it easier to implement a `playNotificationAndRestoreState` function in the future).

**Comment by htilly** [comment]
Awesome! Thanks for the additional Spotify functions. Have been waiting for playlist support! =)

**Pull request merged**